### PR TITLE
feat: support both ESM and CommonJS

### PR DESCRIPTION
**Pull request by ssaarela**

### CODE REVIEW

File: packages/path-parser/package.json [hunk 1]
```diff
@@ -27,7 +33,9 @@
     "parser"
   ],
   "scripts": {
-    "build": "tsc -b ."
+    "build": "yarn build:esm && yarn build:cjs",
+    "build:esm": "tsc -b . && cp ../../package-esm.json dist/esm/package.json",
+    "build:cjs": "tsc -p tsconfig-cjs.json && cp ../../package-cjs.json dist/cjs/package.json"
```
Comment (jlehtiniemi) [inline packages/path-parser/package.json:38]: Same here

File: packages/path/package.json [hunk 2]
```diff
@@ -27,6 +33,8 @@
     "JsonPath"
   ],
   "scripts": {
-    "build": "tsc -b ."
+    "build": "yarn build:esm && yarn build:cjs",
+    "build:esm": "tsc -b . && cp ../../package-esm.json dist/esm/package.json",
+    "build:cjs": "tsc -p tsconfig-cjs.json && cp ../../package-cjs.json dist/cjs/package.json"
```
Comment (jlehtiniemi) [inline packages/path/package.json:38]: Here too

File: packages/core/tsconfig.json [hunk 3]
```diff
@@ -1,8 +1,10 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
-    "outDir": "dist",
-    "rootDir": "src"
+    "rootDir": "src",
+    "module": "esnext",
+    "outDir": "dist/esm",
+    "target": "esnext"
```
Comment (jlehtiniemi) [inline packages/core/tsconfig.json:7]: Should the ESM version of the package use `moduleResolution: node16` or later here?

File: packages/luxon/tsconfig.json [hunk 4]
```diff
@@ -1,8 +1,10 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
-    "outDir": "dist",
-    "rootDir": "src"
+    "rootDir": "src",
+    "module": "esnext",
+    "outDir": "dist/esm",
+    "target": "esnext"
```
Comment (jlehtiniemi) [inline packages/luxon/tsconfig.json:7]: Use `moduleResolution: node16` or later?

File: packages/moment/tsconfig.json [hunk 5]
```diff
@@ -1,8 +1,10 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
-    "outDir": "dist",
-    "rootDir": "src"
+    "rootDir": "src",
+    "module": "esnext",
+    "outDir": "dist/esm",
+    "target": "esnext"
```
Comment (jlehtiniemi) [inline packages/moment/tsconfig.json:7]: Use moduleResolution: node16 or later?

File: packages/path-parser/tsconfig.json [hunk 6]
```diff
@@ -1,8 +1,10 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
-    "outDir": "dist",
-    "rootDir": "src"
+    "rootDir": "src",
+    "module": "esnext",
+    "outDir": "dist/esm",
+    "target": "esnext"
```
Comment (jlehtiniemi) [inline packages/path-parser/tsconfig.json:7]: Use moduleResolution: node16 or later?

File: packages/path/tsconfig.json [hunk 7]
```diff
@@ -1,8 +1,10 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
-    "outDir": "dist",
-    "rootDir": "src"
+    "rootDir": "src",
+    "module": "esnext",
+    "outDir": "dist/esm",
+    "target": "esnext"
```
Comment (jlehtiniemi) [inline packages/path/tsconfig.json:7]: Use moduleResolution: node16 or later?

**Review by Paul53420 — COMMENTED** [review]
Scan through this changes as well for understanding

File: packages/core/tsconfig.json (hunk 3 above)
Comment (ssaarela) [reply packages/core/tsconfig.json:7]: What would be the best option for CJS?

Comment (ssaarela) [reply packages/core/tsconfig.json:7]: ...and is esnext good for module and target?

Comment (ssaarela) [reply packages/core/tsconfig.json:7]: > tsconfig.json(6,15): error TS5110: Option 'module' must be set to 'Node16' when option 'moduleResolution' is set to 'Node16'.

File: packages/luxon/tsconfig.json (hunk 4 above)
Comment (ssaarela) [reply packages/luxon/tsconfig.json]: Couldn't make it work. For proper ES imports in generated files, ES<something> module seems to be required.

**Review by karlkeskyla — APPROVED** [review]
Neat!

**Review by karlkeskyla — APPROVED** [review]

**Pull request merged**